Assets/Scripts/Checkpoint.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+public class Checkpoint : MonoBehaviour
+{
+    [SerializeField]
+    private float inactivatedRotationSpeed = 100, activatedRotationSpeed = 300;
+
+    [SerializeField]
+    private float inactivatedScale = 1, activatedScale = 1.5f;
+
+    [SerializeField]
+    private Color inactivatedColor, activatedColor;
+
+    private bool isActivated = false;
+   // private SpriteRenderer spriterenderer;
+    private AudioSource audioSource;
+
+    private void Start()
+    {
+        //spriterenderer.GetComponent<SpriteRenderer>();
+        audioSource = GetComponent<AudioSource>();
+        UpdateColor();
+    }
+
+    private void Update()
+    {
+        UpdateRotation();
+    }
+
+    private void UpdateColor()
```

Comment:
I see a lot about updating color if a checkpoint is active but since you commented out the spriterenderer, it leads me to believe that you are no longer changing the checkpoints color which is fine but if you are, you should remove the code. If there is something I'm not getting here, then you should at least remove commented out code.